library/include/borealis/core/input.hpp

```diff
@@ -66,11 +69,36 @@ enum ControllerAxis
 };
 
 // Represents the state of the controller (a gamepad or a keyboard) in the current frame
-typedef struct ControllerState
+struct ControllerState
 {
     bool buttons[_BUTTON_MAX]; // true: pressed
     float axes[_AXES_MAX]; // from 0.0f to 1.0f
-} ControllerState;
+};
+
+// Represents a touch phase in the current frame
+enum class TouchPhase
+{
+    START,
+    STAY,
+    END,
+    NONE,
+};
+
+// Contains raw touch data
```

Comment:
Maybe specify the difference here between the two structs?

Something like `Contains raw touch data, filled in by input platform code` and `Contains touch data automatically filled with current phase by the library`